source/material_model/rheology/visco_plastic.cc

```diff
@@ -0,0 +1,893 @@
+
+            // Step 3: strain weakening
+
+            // Step 3a: calculate strain weakening factors for the cohesion, friction, and pre-yield viscosity
+            // If no brittle and/or viscous strain weakening is applied, the factors are 1.
+            const std::array<double, 3> weakening_factors = strain_rheology.compute_strain_weakening_factors(j, in.composition[i]);
+
+            // Step 3b: calculate weakened friction, cohesion, and pre-yield viscosity and adjust the current_stress accordingly
+            const double current_cohesion = drucker_prager_parameters.cohesions[j] * weakening_factors[0];
+            output_parameters.current_friction_angles[j] = drucker_prager_parameters.angles_internal_friction[j] * weakening_factors[1];
+            viscosity_pre_yield *= weakening_factors[2];
+            current_stress *= weakening_factors[2];
+
+            // compute radiation damping if it is used
+            // radiation damping is normally substracted from the shear stress. Here we use current stress instead.
+            // As current stress is only used to compare to yield stress but does not affect material properties,
+            // it is used here to modify current_edot_ii
+            double radiation_damping_term = 0.0;
+            if (friction_options.get_use_radiation_damping())
+              {
+                if (use_elasticity == false)
+                  {
+                    AssertThrow(false, ExcMessage("Usage of radiation damping only makes sense when elasticity is enabled."));
+                  }
+                else
+                  {
+                    // TODO: use the plastic strain rate instead of current_edot_ii
+                    // TODO: more elaborate way to determine cellsize
+                    // TODO: this is not exactly the right way to get the density, as it i only reference densities.
+                    // ideally, you would take the actual densities from out.densities[I],
+                    // i.e. computed as out.densities[i] = MaterialUtilities::average_value (volume_fractions, eos_outputs.densities, MaterialUtilities::arithmetic);
+                    // but at the moment I don't have access to out in this function
+                    const double reference_density = this->get_adiabatic_conditions().density(in.position[0]);
+                    const double cellsize = current_cell->extent_in_direction(0);
+                    //std::cout << " current edot_ii is " << current_edot_ii << std::endl;
+                    radiation_damping_term = current_edot_ii * cellsize * elastic_shear_moduli[j]
+                                             / (2 * sqrt(elastic_shear_moduli[j] / reference_density));
+                    if (yield_mechanism != tresca)
+                      {
+                        current_stress = current_stress - radiation_damping_term;
+                        current_edot_ii = std::max(current_stress / (2 * viscosity_pre_yield), min_strain_rate);
```

Comment:
```suggestion
                        current_edot_ii = std::max(current_stress / (2.0 * viscosity_pre_yield), min_strain_rate);
```
Note to self: should the radiation term act on the stress invariant (shear stress) or on the yield stress?

Reply:
at some point in autumn we talked about that quite often and in the end I applied it to current_stress because current_stress can be used to modify te effective viscosity and current_edot_ii, which in turn modifies the friction angle. Contrary, the yield_stress is only used to be compared to current_stress to determine if we enter yielding, which we always do anyway for RSF. Do you still agree with this reasoning, or do you have another idea?

Reply:
Looking at the Erickson ea 2020 paper, eq 1, they include it in the current stress as well, so let's leave it as is.